File: src/main/java/io/github/hadixlin/iss/mac/MacNative.java
package io.github.hadixlin.iss.mac;

import com.sun.jna.Native;

/** Created by hadix on 30/03/2017. */
public class MacNative {

  static {
    Native.register("input-source-switcher");
  }

  public static native String getCurrentInputSourceID();

  public static native int switchInputSource(String inputSourceID);
}

File: src/main/kotlin/io/github/hadixlin/iss/mac/MacNative.kt
package io.github.hadixlin.iss.mac

import com.sun.jna.Native

/** Created by hadix on 30/03/2017.  */
object MacNative {

	init {
//		val jnaNounpack = System.getProperty("jna.nounpack")
//		System.setProperty("jna.nounpack", "false")
		Native.register("input-source-switcher")
//		if (jnaNounpack != null) {
//			System.setProperty("jna.nounpack", jnaNounpack)
//		}
	}

	external fun getCurrentInputSourceID(): String

	external fun switchInputSource(inputSourceID: String): Int
}